utils/common/src/main/kotlin/EnvironmentVariableFilter.kt

```diff
@@ -0,0 +1,119 @@
+ * process per defaults inherits all environment variables from the parent. This could impose a security risk, for
+ * instance if logic in build scripts could theoretically access sensitive information stored in environment
+ * variables, such as database or service credentials.
+ *
+ * To reduce this list, this object filters the environment variables passed to child processes based on the
+ * following criteria:
+ * - Patterns on variable names can be defined to determine variables with sensitive information. The object provides
+ *   some default patterns to match variables containing strings like "PASS", "USER", "TOKEN", etc.
+ * - There is an allow list to include variables even if they are matched by a pattern.
+ *
+ * So in order to determine whether a specific variable "E" can be passed to a child process, this filter applies the
+ * following steps:
+ * - If E is contained in the allow list, it is included.
+ * - Otherwise, E is included if and only if it is not matched by one of the exclusion patterns.
+ */
+object EnvironmentVariableFilter {
+    /** A list with patterns for variable names that are excluded by default. */
+    val DEFAULT_EXCLUSION_PATTERNS = listOf(
+        "key",
+        "pass",
+        "pwd",
+        "token",
+        "user"
+    )
+
+    /** A list with known variable names that are accepted despite they are matched by exclusion patterns. */
+    val DEFAULT_ALLOW_LIST = setOf(
+        "CARGO_HTTP_USER_AGENT",
+        "COMPOSER_ALLOW_SUPERUSER",
+        "CONAN_LOGIN_ENCRYPTION_KEY",
+        "CONAN_LOGIN_USERNAME",
+        "CONAN_PASSWORD",
+        "CONAN_USERNAME",
+        "CONAN_USER_HOME",
+        "CONAN_USER_HOME_SHORT",
+        "DOTNET_CLI_CONTEXT_ANSI_PASS_THRU",
+        "GIT_ASKPASS",
+        "GIT_HTTP_USER_AGENT",
+        "GRADLE_USER_HOME",
+        "HACKAGE_USERNAME",
+        "HACKAGE_PASSWORD",
+        "HACKAGE_KEY",
+        "PWD",
+        "USER",
+        "USERPROFILE"
+    )
+
+    /** Stores the current exclusion patterns used by this filter. */
+    private var exclusionPatterns = DEFAULT_EXCLUSION_PATTERNS.toPatterns()
+
+    /** Stores the current allow list used by this filter. */
+    private var allowList = DEFAULT_ALLOW_LIST
+
+    /**
+     * Initialize this filter from the given [exclusionPatterns] and [allowList]. The [exclusionPatterns] are
+     * converted to regular expressions, so that they can be matched anywhere in variable names, ignoring case.
+     */
+    fun init(
+        exclusionPatterns: Collection<String> = DEFAULT_EXCLUSION_PATTERNS,
+        allowList: Collection<String> = DEFAULT_ALLOW_LIST
+    ) {
+        this.exclusionPatterns = exclusionPatterns.toPatterns()
+        this.allowList = allowList.toSet()
+    }
+
+    /**
+     * Test whether the variable with the given [name] can be passed to a child process according to the criteria
+     * described in the header comment.
+     */
+    fun acceptVariable(name: String): Boolean {
```

Comment:
Maybe better any of these: `isAccepted`, `canBePassed`, `shouldPropagate` or similar.

Reply:
I liked `isAccepted` best.

Reply:
Actually, with this [other renaming done](https://github.com/oss-review-toolkit/ort/pull/5984#discussion_r1012811315), maybe now better `isAllowed`?

Reply:
Changed to `isAllowed`.